compiler/src/dotty/tools/dotc/transform/init/Objects.scala

```diff
@@ -542,7 +570,7 @@ object Objects:
           refs.map(ref => ref.widen(height)).join
 
         case Fun(code, thisV, klass, env) =>
-          Fun(code, thisV.widen(height), klass, env.widen(height))
+          Fun(code, thisV.widen(height).asInstanceOf[ValueElement], klass, env.widen(height))
```

Comment:
Can we guarantee that `widen` never returns a `RefSet` (and encode that in the return type of `widen`)?